### ISSUE TITLE
More allocations when using LinqGen than normal Linq.

### ISSUE THREAD

**Issue by octoberU** [cathei/LinqGen#4]
Did some testing on a hobby project to see how LinqGen would perform but ended up with slower/more memory heavy results. 

Version used: **v0.2.1**

Default Linq query.
```csharp
value
.OrderBy(x => x.timeMs)
.ThenBy(x => x.behavior)
.ThenBy(x => x.handType)
.ToArray();
```
LinqGen query.
```csharp
value
.Gen()
.OrderBy(new TargetCueSelectors.SelectCueTimeMs())
.ThenBy(new TargetCueSelectors.SelectCueBehavior())
.ThenBy(new TargetCueSelectors.SelectCueHandType())
.ToArray();
```

**Default Linq**
![Unity_stBv7OjpZ2](https://user-images.githubusercontent.com/20176533/212798240-7c64fa77-75b0-482b-a0e3-df7044553abf.png)
**LinqGen**
![Unity_PeVF1SL9zb](https://user-images.githubusercontent.com/20176533/212798243-162208bb-c7ac-41ac-bbb0-95a60e56403c.png)

Is there something I'm doing wrong? The only thing that stands out to me right now is the multiple "ThenBy"s.

**Comment by cathei** [comment]
Thanks for reporting this. This is very interesting. It seems like comparing `int` to `null` in generic comparer causes boxing allocation in Debug setting. If it's in Release setting this consider always false and boxing doesn't happen. The benchmarks are in Release setting so I haven't noticed this. I will update the code to avoid this allocation in Debug setting as well.

**Comment by octoberU** [comment]
Thank you for the quick response, gonna give it another go with the next release once it comes out.

**Comment by cathei** [comment]
It has been included in [v0.2.2](https://github.com/cathei/LinqGen/releases/tag/v0.2.2).
Allocation won't happen in Debug build and editor.
For performance profiling, please use Release mode without deep profiling in the Editor to get correct result.
